scripts/safe-commit-linter.sh

```diff
@@ -54,8 +54,16 @@ fi
 merge_base=`git merge-base "$current_branch_hash" "$MAIN_COMMIT"`
 diff_files=`git diff --name-only "$current_branch_hash" "$merge_base"`
 
+num_files_linted=0
```

Comment:
what does diff_files equal if there is no diff? 
Would be cleaner if you could test for that and do the "No files linted" thing right away rather than using a counter/flag.